### PR TITLE
feat(module:date-picker): support DOWN_ARROW/UP_ARROW/LEFT_ARROW/RIGH…

### LINKED ISSUE #2094: 日期选择的快捷方式不起作用。
## Version
1.3.0
## Environment
Windows 7, Chrome 68
## Reproduction link
[https://ng-zorro-antd-start-abd7zi.stackblitz.io](https://ng-zorro-antd-start-abd7zi.stackblitz.io)
## Steps to reproduce
点击日期控件，按照提示的快捷键选择上一月、下一月，上一年、下一年。
## What is expected?
按照提示的快捷键操作自作用。
## What is actually happening?
上一月下一月(翻页键)，上一年下一年(Control 左右键)，都不起作用。

### PR DESCRIPTION
**Pull request by wilsoncook**
…T_ARROW/PAGE_DOWN/PAGE_UP/ENTER hotkeys

close #2094

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2094

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by wendzhue** [comment]
Some other components like select, cascader have similar behaviors. Should abstract these behaviors in another pr later.

File: components/date-picker/date-picker.component.spec.ts [hunk 1]
```diff
@@ -741,6 +867,11 @@ describe('NzDatePickerComponent', () => {
     fixture.detectChanges();
   }
 
+  function triggerKeydown(node: Node, keyCode: number, ctrlKey: boolean = false, metaKey: boolean = false): void {
+    // tslint:disable-next-line:no-any
+    dispatchEvent(node, new KeyboardEvent('keydown', { keyCode, ctrlKey, metaKey } as any)); // NOTE: don't use `dispatchKeyboardEvent` (not reliable), it will always set `event.metaKey` tobe `true`
```
Comment (JLHwung) [inline components/date-picker/date-picker.component.spec.ts]: s/tobe/to be

File: components/date-picker/lib/popups/date-range-popup.component.ts [hunk 2]
```diff
@@ -98,9 +102,76 @@ export class DateRangePopupComponent implements OnInit, OnChanges {
...
+    this.keyDownSubscription.unsubscribe();
+  }
+
+  onKeyDown(event: KeyboardEvent): void {
+    if (event.target && (event.target as HTMLElement).nodeName.toLowerCase() === 'input') {
+      return ;
+    }
+
+    const ctrlKey = event.ctrlKey || event.metaKey; // NOTE: metaKey is the "⌘ Command" key on `MAC` and the "⊞ Windows" key on `Windows`
+    switch (event.keyCode) {
```
Comment (windyflying) [inline components/date-picker/lib/popups/date-range-popup.component.ts]: Maybe it is a good idea that  the `event.preventDefault()` should be outside the switch block

**Comment by vthinkxie** [comment]
@wilsoncook rebase needed.

**Comment by vthinkxie** [comment]
As there is no update for a long time, the current PR has been closed. If you have any questions, feel free to open another PR, thanks.

**Pull request closed without merging**